criu/clone-noasan.c

```diff
@@ -27,14 +29,80 @@
  *	clone_noasan
  *	... wait for process to finish ...
  *	unlock_last_pid
+ *
+ * This locking is not required if clone3() is available. lock_last_pid()
+ * and unlock_last_pid() will be noops if clone3() is available.
  */
-int clone_noasan(int (*fn)(void *), int flags, void *arg)
+int clone_noasan(int (*fn)(void *), int flags, void *arg, pid_t pid)
 {
 	void *stack_ptr = (void *)round_down((unsigned long)&stack_ptr - 1024, 16);
```

Comment:
Also, what is this?

Reply:
from https://github.com/checkpoint-restore/criu/commit/bcdd276003 :
```
 criu/clone: stack size is too small for Android

stack for clone is too small, child process will get
wild pointer, and segfault.
```
and from https://github.com/checkpoint-restore/criu/commit/ef65d98 :
```
clone_noasan: Allow to create CLONE_VM|CLONE_VFORK processes 
...

Currently parent touches child's stack, as in moment of clone() call
its stack pointer is above the child's (we allocate char stack[128]
on parent's stack). This prevents to create CLONE_VM|CLONE_VFORK
processes, because the child uses stack addresses occupied by parent.

The patch changes clone_noasan() behaviour and allows to do that
with the same memory consumption. We give a child memory, which
is not used by parent clone(), so parent's and child's stacks
have no intersection.
```